src/ui/wxWidgets/wxUtilities.cpp

```diff
@@ -215,3 +215,213 @@ bool IsTaskBarIconAvailable()
 #endif
   return wxTaskBarIcon::IsAvailable();
 }
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// ImagePanel Implementation
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+ImagePanel::ImagePanel(wxPanel *parent, const wxSize &size) : wxPanel(parent, wxID_ANY, wxDefaultPosition, size), m_ImageWidth(), m_ImageHeight(0), m_ImageAspectRatio(0)
+{
+  Bind(wxEVT_PAINT, &ImagePanel::OnPaint, this);
+  Bind(wxEVT_SIZE, &ImagePanel::OnSize, this);
+}
+
+ImagePanel::~ImagePanel()
+{
+  Unbind(wxEVT_PAINT, &ImagePanel::OnPaint, this);
+  Unbind(wxEVT_SIZE, &ImagePanel::OnSize, this);
+}
+
+bool ImagePanel::LoadFromFile(const wxString &file, wxBitmapType format)
+{
+  if (m_Image.LoadFile(file, format)) {
+
+    DetermineImageProperties(m_Image);
+    Refresh(); // Triggers OnPaint to display the image
```

Comment:
I understand that this is where the size problem occurs? If so, did you try refreshing the parent window?

Reply:
I have not yet found a suitable solution.